docs/source/gcp/examples/spanner-set-iam-policy.rst

```diff
@@ -0,0 +1,33 @@
+Spanner - Set IAM policies
+===========================
+
+These policies update the IAM policy for spanner instances and databases, respectively.
+
+.. code-block:: yaml
+
+    policies:
+      - name: gcp-spanner-instances-set-iam-policy
+        resource: gcp.spanner-instance
+        actions:
+          - type: set-iam-policy
+            bindings:
```

Comment:
its very unlikely that a custodian policy will have enough knowledge of a single resource to set the iam policy for it unconditionally. whats more more common is the need to add/remove/update partials

Reply:
doesn't have to be done in this pr, but its good to keep in mind this is really useful without those semantics (which also need a filter on iam policy).

Reply:
@kapilt I have checked setIamPolicy action in GCP. For example https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances/setIamPolicy.
It has "Replaces any existing policy.".
The resource doesn't have information about iamPolicy when list\get methods are invoked. It's available only in getIamPolicy. I am investigating how getIamPolicy can be used in setIamPolicy. If there is a better approach please let me know.

Reply:
first this is mostly awareness, doesn't have to be done in this pr.

second the pattern for custodian is a filter on resource embedded iam policy, that allows filtering the granted permissions and annotates the resource with permission matches. and a set-policy that can remove those matched statements, along with add/modify.

Reply:
@kapilt we updated the PR. Now there is implemented add \ remove modes for setIamPolicy action. Is the implementation valid?

Reply:
cool, i like the direction, so rather than have bindings and mode, it would be nicer to have `add-bindings` and `remove-bindings`, that way it supports adding and removing in a single action.

Reply:
@kapilt we updated the code. Now there are add-bindings and remove-bindings fields in the set-iam-policy action

Reply:
hi @kapilt, anything else to fix in this PR? we have few other cool policy actions in mind which we'd like to inherit from the common class once it's merged